### ISSUE TITLE
Gap-filling operations

### ISSUE THREAD

**Issue by forman** [CCI-Tools/ect-core#81]
Users may want to apply fill gaps in datasets for the following reasons:

* cosmetic: images with gaps don't look good in a map (see #79)
* a subsequent operation requires the input data to be gap-free, e.g. an operation that can't deal with NaN values
* reasonable fill values can be computed so there is no need to let gaps unfilled, which makes a dataset simpler and more user friendly 

Therefore the type of gap-filling and their algorithmic complexity can drastically vary.

Consider:
* spatial (2D) gap-filling
* temporal gap-filling
* multi-dimensional / multi-variate gap-filling (e.g. Kriging)

**Comment by JanisGailis** [comment]
If I remember correctly, during IRM2 we decided upon cosmetic Gap filling only for this iteration. Hence, it's a visualization issue for GUI and plotting ops, not a data manipulation operation. At least for now.

I'll check the minutes.

**Comment by JanisGailis** [comment]
There is some gap-filling code @forman can share.

Otherwise, treat this more as a visualization issue for now.

**Comment by JanisGailis** [comment]
@forman Can you point me to the code you have been talking about? If it works on 2D slices, that should be good enough, I could implement something similar to coregistration then to have spatial_gapfill.

**Comment by forman** [comment]
@JanisGailis  https://github.com/CAB-LAB/gridtools

and the notebook https://github.com/CAB-LAB/gridtools/blob/master/notebooks/gapfill-example.ipynb

**Comment by forman** [comment]
I'd actually like to leave that code there. You may become contributor. Same for the resampling code.

**Comment by JanisGailis** [comment]
I agree. It's conda installable now? So we can include it as a normal dependency?

**Comment by JanisGailis** [comment]
In light of UC11, this is not necessary. Re-projection is done as part of plotting.
